packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,247 @@
+import {
+  Controller,
+  Post,
+  Body,
+  Get,
+  Put,
+  HttpException,
+  HttpStatus,
+  Delete,
+  Param,
+  HttpCode,
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiCreatedResponse,
+  ApiNoContentResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { UserStoreService } from '../../db/user/userStore.service';
+import { HouseStoreService } from '../../db/house/houseStore.service';
+import { Auth } from '../../util/auth.decorator';
+import { TaskUtil } from './tasks.util';
+import { User } from '../../util/user.decorator';
+import { DecodedIdToken } from 'firebase-admin/auth';
+import HouseTasksResponseDto from './dto/house-tasks-response.dto';
+import { CreateTaskDto } from './dto/create-task.dto';
+import { TaskStoreService } from '../../db/task/taskStore.service';
+import { CompleteTaskDto } from './dto/complete-task.dto';
+import UpdateHouseTasksDto from './dto/update-house-tasks.dto';
+import { isValidObjectId } from 'mongoose';
+
+@ApiTags('tasks')
+@Controller('/api/v1/house')
+@Auth()
+export class TasksController {
+  constructor(
+    private readonly houseStoreService: HouseStoreService,
+    private readonly userStoreService: UserStoreService,
+    private readonly taskStoreService: TaskStoreService,
+    private readonly taskUtil: TaskUtil,
+  ) {}
+
+  @Post('/tasks')
+  @ApiOperation({ summary: 'create a new task' })
+  @ApiCreatedResponse({
+    description: 'task created successfully',
+  })
+  @ApiBadRequestResponse({ description: 'Invalid request' })
+  async createTask(
+    @Body() createTaskDto: CreateTaskDto,
+    @User() user: DecodedIdToken,
+  ) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house != undefined) {
```

Comment:
remember that `undefined` is falsy, thus `if(userDoc.house)` should be fine. I've seen this a pattern a few times around repo so if they are still there i think they can be changed

Reply:
ill change them